galoy-client/src/client/mod.rs

```diff
@@ -0,0 +1,233 @@
+            post_graphql::<StablesatsUserLogin, _>(&client, config.api, variables).await?;
+        if response.errors.is_some() {
+            if let Some(error) = response.errors {
+                return Err(GaloyClientError::GrapqQlApi(error[0].clone().message));
+            }
+        }
+
+        let response_data = response.data;
+        if response_data.is_none() {
+            return Err(GaloyClientError::UnknownResponse(format!(
+                "Expected some response data, found {:?}",
+                response_data
+            )));
+        }
+
+        if let Some(response_data) = response_data {
+            let user_login = response_data.user_login;
+            let login_jwt = user_login.auth_token;
+
+            return Ok(GaloyAuthToken { inner: login_jwt });
+        }
+        Err(GaloyClientError::UnknownResponse(
+            "Failed to parse response data".to_string(),
+        ))
+    }
+
+    pub async fn transactions_list(
+        &mut self,
+        last_transaction_cursor: LastTransactionCursor,
+        wallet_currency: stablesats_transactions_list::WalletCurrency,
+    ) -> Result<
+        std::pin::Pin<
+            Box<impl Stream<Item = stablesats_transactions_list::StablesatsTransactionsListMeDefaultAccountTransactionsEdges>>,
+        >,
+        GaloyClientError,
+    >{
+        let variables = stablesats_transactions_list::Variables {
+            last: None,
+            first: None,
+            before: None,
+            after: Some(last_transaction_cursor.inner()),
+        };
+        let response = post_graphql::<StablesatsTransactionsList, _>(
+            &self.client,
+            &self.config.api,
+            variables,
+        )
+        .await?;
+        if response.errors.is_some() {
+            if let Some(error) = response.errors {
+                return Err(GaloyClientError::GrapqQlApi(error[0].clone().message));
+            }
+        }
+
+        let response_data = response.data;
+        let result = match response_data {
+            Some(result) => result,
+            None => {
+                return Err(GaloyClientError::GrapqQlApi(
+                    "Empty `me` in response data".to_string(),
+                ))
+            }
+        };
+        let post_cursor_transactions = StablesatsTransactions::try_from(result)?;
+
+        let tx_edges = post_cursor_transactions.edges;
+
+        let tx = match tx_edges {
+            Some(tx) => tx,
+            None => {
+                return Err(GaloyClientError::GrapqQlApi(
+                    "Empty `transactions edges` in response data".to_string(),
+                ))
+            }
+        };
+
+        let ccy_tx: Vec<stablesats_transactions_list::StablesatsTransactionsListMeDefaultAccountTransactionsEdges> = tx
+            .into_iter()
+            .filter(move |transaction| transaction.node.settlement_currency == wallet_currency)
+            .collect();
+
+        Ok(Box::pin(stream::iter(ccy_tx)))
```

Comment:
I see that you return a stream like I suggested but I see now I made a mistake in my thinking... just return the Vec.

Reply:
This now returns a vector of transactions